### PR TITLE
bugfix: item_filter and Windows ouptut

### PR DESCRIPTION
**Pull request by cadartichaut**
### Short Description:

Fixes the new item_filter which isn't a boolean anymore but a map

Fixed some errors when printing the pokestops names in windows console
### Changes:
- changed bool to map when parsing arguments 
- changed the output's encoding if windows is the used system

@OpenPoGo/maintainers

### CODE REVIEW

**Comment by MaxLeiter** [comment]
:+1: thanks!

[Approved with PullApprove](https://pullapprove.com/OpenPoGo/OpenPoGoBot/pull-request/93/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=OpenPoGo/OpenPoGoBot)

**Comment by meyer9** [comment]
:-1: what is type map and how do you use it?

[Rejected with PullApprove](https://pullapprove.com/OpenPoGo/OpenPoGoBot/pull-request/93/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=OpenPoGo/OpenPoGoBot)

**Comment by cadartichaut** [comment]
my python might be a little rusty, i think i should have used dict here instead of map for a key/value collection

edit: apparently we cant interpret a dictionary with argparse. I had everything in the config file so it was parsed as json later. 

Maybe remove the item_filters from command line and only keep it in the config file?

File: pokemongo_bot/logger.py [hunk 1]
```diff
@@ -26,6 +27,9 @@ def log(string, color='black', fire_event=True):
     output = '[' + time.strftime("%Y-%m-%d %H:%M:%S") + '] ' + string
     if color in color_hex:
         output = b"\033[" + color_hex[color] + output + b"\033[0m"
-    print(output)
+        if platform.system() == "Windows":
+            print(output.encode('cp437', errors='replace').decode('cp437'))
+        else:
+            print(output)
```
Comment (wchill) [inline pokemongo_bot/logger.py]: Incorrectly indented. Also will fail on Python 3 if I remember correctly.

Comment (cadartichaut) [inline pokemongo_bot/logger.py]: can you elaborate on the reason it would fail in python 3 ?

**Comment by meyer9** [comment]
yeah, let's remove it

**Comment by cadartichaut** [comment]
changed it.

Might wanna update the readme as well, it was already out of date regarding the item_filter parameter

**Comment by meyer9** [comment]
I need you to update the README before I can merge this.

File: pokemongo_bot/__init__.py [hunk 2]
```diff
@@ -428,4 +428,4 @@ def get_player_info(self):
                 except KeyError:
                     pass
         except KeyError:
-            pass
+            pass
```
Comment (meyer9) [inline pokemongo_bot/__init__.py]: newline at end of file

**Comment by meyer9** [comment]
newline at end of file please

**Comment by cadartichaut** [comment]
Ok i think we're good now. messed up a few squashes

**Comment by meyer9** [comment]
:+1:

[Approved with PullApprove](https://pullapprove.com/OpenPoGo/OpenPoGoBot/pull-request/93/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=OpenPoGo/OpenPoGoBot)

File: README.md [hunk 3]
```diff
@@ -77,7 +77,7 @@ pip install -r requirements.txt
       -cp COMBAT_POWER, --combat-power COMBAT_POWER Transfer Pokemon that have CP less than this value (default 100)",
       -iv IV, --pokemon-potential IV                Set the ratio for the IV values to transfer (DEFAULT 0.4 eg. 0.4 will transfer a pokemon with IV 0.3)
       -ri, --recycle-items                          Recycle unneeded items automatically
-      -if LIST, --item-filter LIST                  Pass a list of unwanted items to recycle when collected at a Pokestop (e.g, [\"101\",\"102\",\"103\",\"104\"] to recycle potions when collected). Requires --recycle-items. 
+       --item-filter DICT                            Pass a dict of pairs (item: quantity) to recycle when collected at a Pokestop. (e.g {"1": { "keep": 150} will keep up to 150 Pokeballs in the inventory). Requires --recycle-items. Only works in the config file.
```
Comment (MaxLeiter) [inline README.md]: What only works in the config file? --item-filter?

Comment (cadartichaut) [inline README.md]: Yes, item_filters is a dict and we can't interpret dictionaries with argparse, so can't use it in command line

**Comment by kazyka** [comment]
:+1:

File: README.md (hunk 3 above)
Comment (MaxLeiter) [inline README.md]: Then why should we say to pass a dict? We should either have it enable/disable or somehow pass a dictionary.

Comment (cadartichaut) [inline README.md]: I didn't do the change to dictionary, but im guessing it's so people can set a certain amount to keep for each item

Comment (meyer9) [inline README.md]: remove item filter. we already made this decision.

Comment (superyetix) [inline README.md]: remove item filter entirely from the README or is the change I did good enough ?

**Comment by meyer9** [comment]
last thing, just merge this.

**Comment by Reaver01** [comment]
https://github.com/OpenPoGo/OpenPoGoBot/pull/159
will fix this for Windows and FreeBSD. Close this when 159 is merged

**Pull request closed without merging**